docs/features/latency.md

```diff
@@ -0,0 +1,231 @@
+actually present, but many of them are not controllable and many are
+not even measurable. In many cases there are measured values that are
+sums of other values, but ingredients can't be extracted. Values that
+we get at the receiver side are actually two:
+
+* ATS: actual arrival time. It's simply the time when the UDP packet
+has been extracted through the `recvmsg` system call.
+
+* TS: time recorded in the packet header, set on the sender side and extracted
+from the packet at the receiver side
+
+Note that the timestamp in the packet's header is 32-bit, which gives
+it more-less 2.5 minutes to roll over. Therefore there is the timestamp
+rollover tracked and a segment increase is done in order to keep an
+eye on the overall actual time. For the needs of the formula definitions
+it will be stated that TS is the true difference between the connection
+start time and the time when the sending time has been declared when
+the sender application is calling any of the `srt_send*` functions
+(see [`srt_sendmsg2`](../API/API-functions.md#srt_sendmsg2) for details).
+
+
+SRT latency components
+======================
+
+To understand the latency components we need also other definitions:
+
+* ETS: expected arrival time. This is the time of the packet when it
+"should" arrive according to its TS
+
+* PTS: packet's play time. It's the time when SRT gives up the packet
+to the `srt_recvmsg2` call (that is, it sets up the IN flag in epoll
+and resumes the blocked function call, if it was in blocking mode).
+
+* STS: the time declared as the sending time when the packet was
+scheduled for sending at the sender side (if you don't use the
+declared time, by default it's the monotonic time taken when this
+function was called), which is represented by TS.
+
+* RTS: the same as STS, but calculated at the receiver side. The
+only way to extract it is by using some initial statements.
+
+The "true latency" for a particular packet in SRT can be simply defined as:
+
+* `TD = PTS - STS`
+
+Note that this is a stable definition (independent on the packet),
+but this value is not really controllable. So let's define the PTS
+for the packet `x`:
+
+* `PTS[x] = ETS[x] + LATENCY + DRIFT`
+
+where `LATENCY` is the negotiated latency value (out of the
+`SRTO_RCVLATENCY` on the agent and `SRTO_PEERLATENCY` on the peer)
+and DRIFT will be described later (for simplification you can
+state it's initially 0).
+
+These components undergo the following formula:
+
+* `ETS[x] = start_time + TS[x]`
+
+Note that it's not possible to simply define it basing on STS
+because sender and receiver are two different machines that can only
+see one another through the network, but their clocks are separate,
+and can even run on different or changing speeds, while the only
+visible phenomena happen only at a packet arrival machine. This
+above formula, however, allows us to define the start time because
+we state the following for the very first data packet:
```

Comment:
```suggestion
Note that it's not possible to simply define a "true latency based on STS
because the sender and receiver are two different machines that can only
see one another through the network. Their clocks are separate,
and can even run at different or changing speeds, and the only
visible phenomena happens when packets arrive at the receiver machine. 
However, the formula above does allow us to define the start time because
we state the following for the very first data packet:
```

Reply:
I've let myself fix some typos.

Reply:
Oof . . . sorry I let those slip in. Thanks for catching them!